simphony/io/file_lattice.py

```diff
@@ -0,0 +1,378 @@
+            newtable.attrs.base_vect = \
+                copy.deepcopy(self._table.attrs.base_vect)
+            newtable.attrs.size = copy.deepcopy(self._table.attrs.size)
+            newtable.attrs.origin = copy.deepcopy(self._table.attrs.origin)
+
+            # Delete old table and rename new table
+            self._table._f_remove()
+            self._table = newtable
+            self._file.rename_node(self._group, self._name, "tmp")
+
+            # Check if larger bitmask has to be generated
+            if len(self._ColDescr.columns.keys()) > self._mask_size:
+                self._mask_size = len(self._ColDescr.columns.keys())
+                atom = tables.BoolAtom(shape=(self._mask_size))
+                filters = tables.Filters(complevel=5, complib='zlib')
+                self._newbitmask = \
+                    self._file.create_carray(self._group,
+                                             self._maskname+'_new', atom,
+                                             self._size, filters=filters)
+
+                for n in xrange(numpy.prod(self._bitmask.shape)):
+                    self._newbitmask[self._index(n)][0:self._bitmask.atom.shape[0]] = \
+                        self._bitmask[self._index(n)]
+
+                self._bitmask._f_remove()
+                self._bitmask = self._newbitmask
+                self._file.root.lattice._v_leaves[
+                    self._maskname+'_new'].rename(self._maskname)
+
+    def _update_bitmask(self, id, name):
+        """ Writes True to bitmask to flag a defined DataContainer for
+        lattice_node at position 'id'
+
+        Parameters:
+            id : tuple
+                LatticeNode coordinates
+            name : str
+                Name of the PyTables column that is flagged
+        """
+        bits = self._bitmask[id]
+        n = self._ColDescr.columns[name]._v_pos
+        newbits = numpy.array(numpy.zeros(self._mask_size), dtype=bool)
+        newbits[n] = True
+        self._bitmask[id] = bits | newbits
+
+    def update_node(self, node, keywords=[]):
+        """ Updates DataContainer contents for LatticeNode node
+
+        Parameters:
+            node : LatticeNode
+                reference to LatticeNode object
+            keywords : list, optional
+                List of CUBA-keywords that are updated. If empty list
+                then all DC's are from the node
+        """
+        # Find correct row for lattice_node
+        n = self._get_row_index(node.id)
+
+        # Check that table has all the needed columns
+        if keywords == []:
+            # Add everything
+            self._update_table(node, node.data.keys())
+            for row in self._table.iterrows(n, n+1):
+                if len(node.data) is not 0:
+                    for key in node.data:
+                        name = str(key)[5:]
+                        row[name] = copy.deepcopy(node.data[key])
```

Comment:
There should be no need for deep copy or even copy since the data will be converted/copied into the column data type as necessary.